src/main/java/org/terasology/simpleFarming/systems/BushAuthoritySystem.java

```diff
@@ -155,12 +155,31 @@ public void onBushGrowth(DelayedActionTriggeredEvent event, EntityRef bush, Bush
      * @param bushComponent the definition of the bush being grown
      * @param stages        the number of stages to grow; negative values represent un-growth
      */
-    private EntityRef doBushGrowth(BushDefinitionComponent bushComponent, int stages) {
-        bushComponent.currentStage += stages;
-        worldProvider.setBlock(bushComponent.position, bushComponent.stages[bushComponent.currentStage].block);
-        EntityRef newBush = blockEntityRegistry.getBlockEntityAt(bushComponent.position);
-        newBush.addOrSaveComponent(bushComponent);
-        return newBush;
+    private void doBushGrowth(Vector3i position, BushDefinitionComponent bushComponent, int stages) {
+        EntityRef newBush = null;
+        Map.Entry<String, GrowthStage> stage = null;
+        if (!isInLastStage(bushComponent) || stages < 0) {
+            bushComponent.currentStage += stages;
+            stage = getGrowthStage(bushComponent, bushComponent.currentStage);
+            worldProvider.setBlock(position, blockManager.getBlock(stage.getKey()));
+            newBush = blockEntityRegistry.getBlockEntityAt(position);
+            newBush.addOrSaveComponent(bushComponent);
+        }
+        if( newBush != null && stage.getValue().maxTime > 0 && stage.getValue().minTime > 0) {
```

Comment:
Unless I'm mistaken, the only time `newBush` is not `null` is if the preceding `if` is executed. This should be nested inside the other `if`, also meaning that `newBush` and `stage` don't need to be initialised/declared outside the if statement.

Additionally, handling for the `maxTime` is done in the `generateRandom` method.

Reply:
Yes!  Good insight.

I have not seen handling of maxTime when both max and min are 0 (which happens if they are not specified in json).  Maybe there is a better way to address this than the double >0 check?